### PR TITLE
Centralize non-JSON considerations

### PR DESCRIPTION
**Pull request by cabo**
Centralize the text that explains how non-interoperable JSON may cause unpredictable behavior.

This is a little clumsy as the sections to be referenced are currently unnumbered.

### CODE REVIEW

**Comment by cabo** [comment]
Solves a few "behaviours", too :-)

File: draft-ietf-jsonpath-base.md [hunk 1]
```diff
@@ -439,12 +439,25 @@ errors.)
...
+The present specification does not attempt to define predictable
+behavior for JSONPath queries in these situations.
+(Note that another warning about interoperability, in {{Section 2 of
+-json}}, at the time of writing is generally considered to be overtaken
+by events and causes no issues with the present specification.)
+
+Specifically, the "Semantics" subsections of Sections {{<dot-selector}},
+{{<wildcard}}, {{<index-selector}}, {{<index-wildcard-selector}},
+{{<filter-selector}}, and {{<descendant-selectors}} describe behavior that
+turns into unpredictable when the JSON value for one of the objects
```
Comment (glyn) [inline draft-ietf-jsonpath-base.md:454]:
```suggestion
turns unpredictable when the JSON value for one of the objects
```

**Review by glyn — CHANGES_REQUESTED** [review]
One suggestion. Other than that, I like the approach as it is less "fussy".

**Review by glyn — APPROVED** [review]

**Comment by goessner** [comment]
This centralizing approach is a good thing.

I suggest to reuse the phrase "duplicate names" for better recognition of the wording in RFC8259, though

> `multiple members for a single object that share the same member name`

is more general, as it addresses more than two.

File: draft-ietf-jsonpath-base.md [hunk 2]
```diff
@@ -439,12 +439,25 @@ errors.)
...
+-json}}, at the time of writing is generally considered to be overtaken
+by events and causes no issues with the present specification.)
+
+Specifically, the "Semantics" subsections of Sections {{<dot-selector}},
+{{<wildcard}}, {{<index-selector}}, {{<index-wildcard-selector}},
+{{<filter-selector}}, and {{<descendant-selectors}} describe behavior that
+turns unpredictable when the JSON value for one of the objects
+under consideration was constructed out of JSON text that exhibits
+multiple members for a single object that share the same member name
+({{Section 4 of -json}}), and matching ({{selectors}}) and comparing
```
Comment (cabo) [inline draft-ietf-jsonpath-base.md:457]: Re Stefan's suggestion to include the phrase "duplicate names":

```suggestion
("duplicate names", see {{Section 4 of -json}}), and matching ({{selectors}}) and comparing
```

File: draft-ietf-jsonpath-base.md [hunk 3]
```diff
@@ -439,12 +439,25 @@ errors.)
...
+Specifically, the "Semantics" subsections of Sections {{<dot-selector}},
+{{<wildcard}}, {{<index-selector}}, {{<index-wildcard-selector}},
+{{<filter-selector}}, and {{<descendant-selectors}} describe behavior that
+turns unpredictable when the JSON value for one of the objects
+under consideration was constructed out of JSON text that exhibits
+multiple members for a single object that share the same member name
+("duplicate names", see {{Section 4 of -json}}), and matching ({{selectors}}) and comparing
+({{comparisons}} in Section {{<filter-selector}}) of strings assumes these
+strings are sequences of Unicode scalar values, turning unpredictable
+if they aren't ({{Section 8.2 of -json}}).
```
Comment (glyn) [inline draft-ietf-jsonpath-base.md:460]: I'm finding it hard to parse this sentence from "and matching" onwards. I suggest splitting into two or more sentences.

**Review by glyn — COMMENTED** [review]
The English in the overview is a bit tortuous.

File: draft-ietf-jsonpath-base.md [hunk 4]
```diff
@@ -439,12 +439,25 @@ errors.)
...
+-json}}, at the time of writing is generally considered to be overtaken
+by events and causes no issues with the present specification.)
+
+Specifically, the "Semantics" subsections of Sections {{<dot-selector}},
+{{<wildcard}}, {{<index-selector}}, {{<index-wildcard-selector}},
+{{<filter-selector}}, and {{<descendant-selectors}} describe behavior that
+turns unpredictable when the JSON value for one of the objects
+under consideration was constructed out of JSON text that exhibits
+multiple members for a single object that share the same member name
+("duplicate names", see {{Section 4 of -json}}), and matching ({{selectors}}) and comparing
```
Comment (cabo) [inline draft-ietf-jsonpath-base.md:457]:
```suggestion
("duplicate names", see {{Section 4 of -json}}).
Also, matching ({{selectors}}) and comparing
```

File: draft-ietf-jsonpath-base.md (hunk 3 above)
Comment (cabo) [reply draft-ietf-jsonpath-base.md:460]: (see suggestion above)

**Review by goessner — APPROVED** [review]

**Review by glyn — APPROVED** [review]
There may be scope for fine-tuning the section cross-referencing, but this can be deferred to purely editorial activity
